### ISSUE TITLE
S2583 bool? false positive

### ISSUE THREAD

**Issue by wrightsonm** [SonarSource/sonar-dotnet#2530]
### Description

if (result ?? false)  always returns the S2583 warning

Warning	S2583	Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Repro steps

```
[TestMethod]
        public void S2583()
        {
            bool? result = S2583_Test();
            if (result ?? false)
            {
                Assert.IsNull(result);
            }
        }
        private bool? S2583_Test()
        {
            return null;
        }
```

### Expected behavior

S2583 is not raised

### Actual behavior

S2583 is raised

### Known workarounds

Change it to a two line version and convert to a bool first

```
bool? result = S2583_Test();
            bool e = result ?? false;
            if (e)
            {
                Assert.IsNull(result);
            }
```

### Related information

* SonarC# Version SonarLint for Visual Studio 4.11.1.0
* Visual Studio Version 2017 15.9.6
* If running through the Scanner for MSBuild, its version N/A

**Comment by andrei-epure-sonarsource** [comment]
thanks @wrightsonm , this seems to be a duplicate of #2369 . This will be fixed in the next release of `SonarCsharp` (and one of the next releases of `sonarlint-visualstudio`).
